master/buildbot/reporters/mail.py

```diff
@@ -43,6 +43,11 @@
 from buildbot.reporters.message import MessageFormatter as DefaultMessageFormatter
 from buildbot.util import service
 
+if PY3:
```

Comment:
StringIO doesn't work the same on Python 2 and Python 3, since in Python 2, strings are bytes, and in Python 3 strings are unicode.
If I follow your advice, the code will break on Python 2, and clients which use the code will need to change to use.

That's why I used the technique I did.


While it is useful to look at the advice from the future author,
you also need to understand the code and do the right thing, and not blindly
follow advice.